src/batch-audio-transcription/Program.cs

```diff
@@ -0,0 +1,226 @@
+                        }
+                        // We have the parsed arguments, so let's just pass them down
+                        return await ProcessMediaFiles(opts.Input, opts.Locale, opts.Output, opts.Key, opts.Region, opts.Verbose);
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine($"Error! {ex.Message}");
+                        return -3; // Not handled error
+                    }
+                },
+               errs => Task.FromResult(-1) // Invalid arguments
+              );
+        }
+
+        private static async Task<int> ProcessMediaFiles(string inputFolder, string locale, string output, string speechServiceKey, string speechServiceRegion, bool verbose)
+        {
+            // Create the speeach config we are going to use for the files
+            var config = SpeechConfig.FromSubscription(speechServiceKey, speechServiceRegion);
+            config.SpeechRecognitionLanguage = locale;
+            config.OutputFormat = OutputFormat.Detailed;
+
+            var results = new List<FileResult>();
+            var allGood = true;
+            // The for each file loop
+            var fileNames = Directory.GetFiles(inputFolder, "*.*", SearchOption.AllDirectories);
+            foreach (var fileName in fileNames)
+            {
+                // Call the ProcessMediaFile
+                try
+                {
+                    var audiofileresult = new FileResult(fileName);
+                    results.Add(audiofileresult);
+                    audiofileresult.FileLoaded = await ProcessMediaFile(config, audiofileresult, verbose);
+                    if (!audiofileresult.FileLoaded)
+                    {
+                        Console.WriteLine($"Could not load file '{fileName}' as an audio file.");
+                        Console.WriteLine("Try installing missing codecs, like gstreamer.");
+                        allGood = false;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // Catch exception in case it's not an audio file or it's corrupted.
+                    Console.WriteLine($"Error processing file {fileName}!");
+                    Console.WriteLine($"Error: {ex.Message}");
+                    allGood = false;
+                }
+            }
+
+            // Create an excel file and save the results
+            SaveResultsInXL(output, results);
+
+            return allGood ? 0 : -4;
+        }
+
+        private static void SaveResultsInXL(string output, List<FileResult> results)
+        {
+            // We are using the https://closedxml.github.io/ClosedXML/
+            // for excel manipulation in .net 6
+            using (var workbook = new XLWorkbook())
+            {
+                var worksheet = workbook.Worksheets.Add("TranscriptionResults");
+               
+                DataTable table = new DataTable();
```

Comment:
Perhaps we can change this in another version to use the LINQ approach described in the docs https://github.com/closedxml/closedxml/wiki/Inserting-Tables